Server/Controllers/PostsController.cs

```diff
@@ -0,0 +1,204 @@
+            }
+        }
+        [HttpPut("likes/{id}")]
+        public IActionResult AddLike(int id, [FromBody] Post post)
+        {
+            try
+            {
+                using (fbContext fb = new fbContext())
+                {
+                    var update = fb.Posts.First(x => x.PostId == id);
+                    update.Likes = post.Likes + 1;
+                    fb.SaveChanges();
+                    return Ok(update.Likes);
+                }
+            }
+            catch (Exception)
+            {
+                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            }
+        }
+        [HttpPut("dislikes/{id}")]
+        public IActionResult UnLike(int id, [FromBody] Post post)
+        {
+            try
+            {
+                using (fbContext fb = new fbContext())
+                {
+                    var update = fb.Posts.First(x => x.PostId == id);
+                    if (post.Likes != 0)
+                    {
+                        update.Likes = post.Likes - 1;
+                        fb.SaveChanges();
+
+                        return Ok(update.Likes);
+                    }
+                    return Ok(0);
+                }
+            }
+            catch (Exception)
+            {
+                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            }
+        }
+
+        [HttpPut("hearts/{id}")]
+        public IActionResult AddHeart(int id, [FromBody] Post post)
+        {
+            try
+            {
+                using (fbContext fb = new fbContext())
+                {
+                    var update = fb.Posts.First(x => x.PostId == id);
+                    update.Hearts = post.Hearts + 1;
+                    fb.SaveChanges();
+                    return Ok(update.Hearts);
+                }
+            }
+            catch (Exception)
+            {
+                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            }
+        }
+
+        [HttpPut("dishearts/{id}")]
+        public IActionResult RemoveHeart(int id, [FromBody] Post post)
+        {
+            try
+            {
+                using (fbContext fb = new fbContext())
+                {
+                    var update = fb.Posts.First(x => x.PostId == id);
+                    if (post.Hearts != 0)
+                    {
+                        update.Hearts = post.Hearts - 1;
+                        fb.SaveChanges();
+                        return Ok(update.Hearts);
+                    }
+
+                    return Ok(0);
+
+
+                }
+            }
+            catch (Exception)
+            {
+                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            }
+        }
+
+        // DELETE api/<PostsController>/5
+        [HttpDelete("{id}")]
+        public IActionResult RemovePost(int id)
```

Comment:
delete this endpoint as per story we don't have delete post

Reply:
deleted the endpoint